cmake/Modules/DefaultOptions.cmake

```diff
@@ -1,3 +1,23 @@
+set(MR_CXX_STANDARD 20 CACHE STRING "Version of the C++ standard used to compile the project")
+IF(DEFINED ENV{MR_USE_CPP_23} AND "$ENV{MR_USE_CPP_23}" STREQUAL "ON")
+  message("MR_USE_CPP_23 variable is deprecated; consider setting MR_CXX_STANDARD to 23")
+  set(MR_CXX_STANDARD 23)
+ENDIF()
+set(CMAKE_CXX_STANDARD ${MR_CXX_STANDARD})
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -D_DEBUG -DDEBUG")
+# turn on warnings as errors
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-missing-field-initializers -Wno-unknown-pragmas -Wno-sign-compare -Werror -fvisibility=hidden -pedantic-errors -DIMGUI_DISABLE_OBSOLETE_FUNCTIONS -DIMGUI_ENABLE_FREETYPE")
+
+IF(WIN32 AND MINGW)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wa,-mbig-obj")
+ENDIF()
+
+IF(NOT MR_EMSCRIPTEN_SINGLETHREAD)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+ENDIF() # NOT MR_EMSCRIPTEN_SINGLETHREAD
```

Comment:
should it be under global `IF( MR_EMSCRIPTEN )` ?

Reply:
The flag is required for other platforms as well.